homeassistant/components/light/group.py

```diff
@@ -0,0 +1,262 @@
+        return self._white_value
+
+    @property
+    def effect_list(self) -> Optional[List[str]]:
+        """Return the list of supported effects."""
+        return self._effect_list
+
+    @property
+    def effect(self) -> Optional[str]:
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self._supported_features
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed for a group light."""
+        return False
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Forward the turn_on command to all lights in the group."""
+        for entity_id in self._entity_ids:
+            kwargs[ATTR_ENTITY_ID] = entity_id
+            yield from self.hass.services.async_call(
+                'light', SERVICE_TURN_ON, kwargs, blocking=True)
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Forward the turn_off command to all lights in the group."""
+        for entity_id in self._entity_ids:
+            kwargs[ATTR_ENTITY_ID] = entity_id
+            yield from self.hass.services.async_call(
+                'light', SERVICE_TURN_OFF, kwargs, blocking=True)
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Query all members and determine the group state."""
+        states = self._child_states()
+
+        self._state = _determine_on_off_state(states)
+        self._brightness = _reduce_attribute(states, 'brightness')
+        self._xy_color = _reduce_attribute(states, 'xy_color')
+        self._rgb_color = _reduce_attribute(states, 'rgb_color')
+        self._color_temp = _reduce_attribute(states, 'color_temp')
+        self._min_mireds = _reduce_attribute(
+            states, 'min_mireds', default=154, force_on=False)
+        self._max_mireds = _reduce_attribute(
+            states, 'max_mireds', default=500, force_on=False)
+        self._white_value = _reduce_attribute(states, 'white_value')
+
+        all_effect_lists = list(
+            _find_state_attributes(states, 'effect_list', force_on=False))
+        self._effect_list = None
+        if all_effect_lists:
+            # Merge all effects from all effect_lists with a union merge.
+            self._effect_list = list(set().union(*all_effect_lists))
+
+        all_effects = list(_find_state_attributes(states, 'effect'))
+        self._effect = None
+        if all_effects:
+            flat_effects = list(itertools.chain(*all_effect_lists))
+            # Report the most common effect.
+            self._effect = max(set(flat_effects), key=flat_effects.count)
+
+        self._supported_features = 0
+        for support in _find_state_attributes(
+                states, 'supported_features', force_on=False):
+            # Merge supported features by emulating support for every feature
+            # we find.
+            self._supported_features |= support
```

Comment:
I would expect `&=` ?

Reply:
I would expect that too. (as long as we're able to manually override the supported features)
De default should be to only support features that all child entities support (bitwise &=)

But there may be instances where you have some lights that do color, and some do only color temperature. If you group those you want to at least be able to set the color temperature. (And let the platform of the colored light figure out how to handle that further.)
To accomplish that you need to be able to manually override supported_features. But that might be food for another PR.... let's not make this one too complicated maybe

Reply:
I'm creating a architecture post right now, as I think we should first define how users expect these platforms to behave.